Final merge candidate for responsiveness, core web vitals, and css refactoring

jmsrpp opened the pull request:
## What reference architecture does this PR apply to?
This is the successor for #406 with following high-level enhancements:
1. Core web vitals improved through image optimization, CLS protection, hydration guards
2. Moved FilterBar, ReactCarousel, and HighlightText to shared components
3. Refactored all swizzled components to use design-and-styling.md best practices

## Who should review your contribution? (Use @mention)
@navyakhurana @cernus76 @johannagonnzdz @julian-schambeck 

## Checklist before submitting
- [x] My commits are only for the reference architecture mentioned above.
- [x] I have followed the folder structure in the [main README](../README.md)

julian-schambeck reviewed (commented):
did some more testing: in dark mode the dots right under the carousel aren't visible.

on the card-style overview, selecting two technology domains causes a weird visual (see below). is it realistic that on ref arch belongs to _two_ domains? if not, maybe we should limit it to one filter selection.

<img width="1134" height="955" alt="image" src="https://github.com/user-attachments/assets/69f9f23c-3501-417c-b711-9156455417e3" />

jmsrpp commented:
> did some more testing: in dark mode the dots right under the carousel aren't visible.
> 
> on the card-style overview, selecting two technology domains causes a weird visual (see below). is it realistic that on ref arch belongs to _two_ domains? if not, maybe we should limit it to one filter selection.
> 
Thanks @julian-schambeck for the details. I fixed these issues in the following way:
1. Add global styles for the react-slick buttons. I tried to do it in the module css but since these are scoped to the module instance, the global styles from the npm module superseded. Setting it in custom.css resolved this.
2. I agree with >1 domain or partner. This should be an OR and not an AND. So I changed the filter criteria from .every to .some.
3. Now you always get results, but in case we add more partners or domains in the future, it is possible there will be no results. So I implemented a small No Results container to handle it. You can test (if you want) by changing the filter clause on L210 and 214 back to every.

julian-schambeck commented:
got it. tested again and i can see that it works, including the 'no results container'. good to go from my side :)

jmsrpp commented:
> Hey, I have seen a couple of regressions compared to PR #406. I paste a few findings here but let's have a little chat once you are in WDF.
> 
Thanks @cernus76 for finding these. I made some updates to fix the alignment, ensure the proper order of title and button controls, and unpicked a regression where we were re-using the default container class on the community section. This caused some responsive misalignment in the explore page.

As I have modified the button controls and normalized use of the sectionContainer across all carousel implementations, we must expect some small changes like the space at the top of the carousel component. I have made this configurable so we can change it. Let's discuss on Monday and I will place this back in ready for review, as we really need to address the CLS and LCP stats.

cernus76 reviewed (approved):
Good to go!
